homeassistant/components/twinkly/light.py

```diff
@@ -180,6 +181,13 @@ async def async_added_to_hass(self) -> None:
                     self.supported_features & ~LightEntityFeature.EFFECT
                 )
 
+            device_registry = dr.async_get(self.hass)
+            device_entry = device_registry.async_get_device({(DOMAIN, self._id)}, set())
+            if device_entry:
+                device_registry.async_update_device(
+                    device_entry.id, sw_version=self._software_version
+                )
+
```

Comment:
Seems unrelated to this PR

Reply:
I added those lines to get the firmware version into the Device page as well.  But I can remove it and add another PR for that if you think that is better.

Reply:
It seems as software version is already part of DeviceInfo so it should already be there?

Reply:
I don't know exactly how this is supposed to work, but when I follow the logs, it looks like the call to device_info() only happens once, and it happens before added_to_hass() (whch is where the software version is fetched).

If I should get the software version at another stage, I am more than happy to fix that instead.

Reply:
I think it should update anyway but I see it's set empty at init at first and later updated which perhaps makes the empty string persistent.
You might want to check .storage/device_registry file if it's blank or actually the right sw_version (without applying this PR obviously).